test/VoteManager.js

```diff
@@ -533,6 +535,192 @@ describe('VoteManager', function () {
+        const epoch = await getEpoch();
+        const stakerIdAcc7 = await stakeManager.stakerIds(signers[7].address);
+        const maxAssetsPerStaker = Number(await parameters.maxAssetsPerStaker());
+        const numAssets = Number(await assetManager.getNumAssets());
+
+        const votes = [0, 0, 0, 0, 0, 0, 0, 0, 0];
+        const tree = merkle('keccak256').sync(votes);
+
+        await mineToNextState(); // reveal
+
+        const proof = [];
+        for (let i = 0; i < votes.length; i++) {
+          proof.push(tree.getProofPath(i, true, true));
+        }
+
+        const assignedAssets = await getAssignedAssets(numAssets, stakerIdAcc7, votes, proof, maxAssetsPerStaker, random);
+        const assigneedAssetsVotes = assignedAssets[0];
+        const assigneedAssetsProofs = assignedAssets[1];
+
+        const tx = voteManager.connect(signers[7]).reveal(epoch, tree.root(), assigneedAssetsVotes, assigneedAssetsProofs,
+          '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd',
+          signers[7].address);
+        // arguments getvVote => epoch, stakerId, assetId
+
+        await assertRevert(tx, 'revert');
+
+        // To Form Block Proposal On basis of revealed assets this epoch
+        revealedAssetsThisEpoch = {};
+        for (let i = 0; i < maxAssetsPerStaker; i++) {
+          revealedAssetsThisEpoch[assigneedAssetsVotes[i].id] = true;
+        }
+
+        blockThisEpoch = {
+          ids: [], medians: [],
+        };
+        for (let i = 1; i <= numAssets; i++) {
+          if (revealedAssetsThisEpoch[i]) {
+            blockThisEpoch.ids.push(i);
+            blockThisEpoch.medians.push(i * 0);
+          }
+        }
+      });
+      it('if the reported value is zero, should not be able to propose', async function () {
+        const epoch = await getEpoch();
+
+        await mineToNextState(); // propose
+        const stakerIdAcc7 = await stakeManager.stakerIds(signers[7].address);
+        const staker = await stakeManager.getStaker(stakerIdAcc7);
+
+        const { biggestInfluencerId } = await getBiggestInfluenceAndId(stakeManager);
+        const iteration = await getIteration(stakeManager, random, staker);
+
+        const tx = blockManager.connect(signers[7]).propose(epoch,
+          blockThisEpoch.ids,
+          blockThisEpoch.medians,
+          iteration,
+          biggestInfluencerId);
+
+        assertRevert(tx, 'revert');
+      });
+
+      it('if the reported value is zero, should not be able to dispute', async function () {
+        await mineToNextState(); // dispute
+        const epoch = await getEpoch();
+
+        const sortedVotes = [toBigNumber(toBeDisputedAssetId * 0)];
+
+        const tx = blockManager.connect(signers[8]).giveSorted(epoch, toBeDisputedAssetId, sortedVotes);
+
+        await assertRevert(tx, 'sorted[i] is not greater than lastVisited');
+      });
+
+      it('if the reported value is zero , nothing should work in next epoch', async function () {
```

Comment:
why nothing should work in next epoch?

Reply:
Will rename the testCase name to "if reported value is zero , only commit should work in next epoch " 

Does this sound good ?

Reply:
Will rename the testCase name to "if reported value is zero , only commit should work in next epoch " 

Does this sound good ?

Reply:
I dont understand this testcase? You are writing a test case that should pass because of a bug existing in the code.

Reply:
No, I am only checking that only commit should work in each epoch if the reported value is zero but as of now we don't have any require in reveal for the reported value .


In both these testCases I have written positive test cases for commit and negative testCases for remaining of the cycle (reveal, propose, dispute ).

Reply:
please add require() statements too to fix the bug

Reply:
#259

Reply:
Sure.

Reply:
> I dont understand this testcase? You are writing a test case that should pass because of a bug existing in the code.

@GaurShekhar "Nothing should work in next epoch" is a little bit vague. Please write a description that does not have ambiguity.